pacta_data_qa.Rmd

```diff
@@ -0,0 +1,660 @@
+params: 
+  project_code: general
+  project_location_ext: NULL
+  twodii_internal: TRUE
+  project_name: NULL
+output:
+  html_document:
+    fig_caption: yes
+    theme: flatly
+    toc: yes
+    toc_depth: 3
+    toc_float: yes
+---
+
+```{r setup, include=FALSE}
+# set echo = TRUE, for more insight to the code printed in the output
+knitr::opts_chunk$set(echo = FALSE)
+library(tidyverse)
+library(DT)
+
+project_code <- params$project_code
+project_location_ext <- params$project_location_ext
+twodii_internal <- params$twodii_internal
+project_name <- params$project_name
+
+create_project_folder(project_name, twodii_internal, project_location_ext)
+set_project_paths(project_name, twodii_internal, project_location_ext)
+
+```
+
+## Use case
+This document can be used to perform Data QA on processed Portfolio Input files for PACTA projects.
+The aim of this analysis is to ensure data consistency and find holdings, that may not be considered in the PACTA analysis yet, but could be included in one of the below described ways in order to increase portfolio coverage. 
+
+## Read in data
+
+For the analysis to be run, the *processed portfolio inputs* as well as the *company level results* for both corporate bonds and listed equity need to be read in. Given the normal PACTA project structure is used, this .Rmd will automatically locate the corresponding files and read them in. In case the folder structure of the project was changed, the paths need to be adjusted manually (this should rarely be the case).
+
+```{r}
+# read in if report has been run successfully, but data is not currently in the workspace
+# availability checks
+bonds_available <- c("Bonds_results_company.rda") %in% list.files(results_path)
+equity_available <- c("Equity_results_company.rda") %in% list.files(results_path)
+portfolio_available <- c(paste0(project_name, "_total_portfolio.rda")) %in% list.files(paste0(proc_input_path, "/"))
+# data paths
+# inconsisten directory with the check (Meta Investor sub dir)
+portfolio_path <- paste0(project_location,"/30_Processed_Inputs/",project_name,"_total_portfolio.rda")
+bonds_path <- paste0(project_location,"/40_Results/Meta Investor/Bonds_results_company.rda")
+equity_path <- paste0(project_location,"/40_Results/Meta Investor/Equity_results_company.rda")
+
+# read in as much asset types as possible
+if (all(bonds_available, equity_available, portfolio_available)) {
+  result_data_qa <- readRDS(portfolio_path)
+  bonds_results_company <- readRDS(bonds_path)
+  equity_results_company <- readRDS(equity_path)
+} else if (all(bonds_available, portfolio_available)) {
+  result_data_qa <- readRDS(portfolio_path)
+  bonds_results_company <- readRDS(bonds_path)
+} else if (all(equity_available, portfolio_available)) {
+  result_data_qa <- readRDS(portfolio_path) &
+  equity_results_company <- readRDS(equity_path)
+} else {
+  print("Could not find the required files to generate data QA html document in the project directory. Skipping!")
+}
```

Comment:
It's a bit confusing why these different branches are needed. I would test if you have the minimum requirements, and if not print the message (and also maybe error out?). Then for each file, if it's available, load it.